crates/swc_ecma_codegen/src/lib.rs

```diff
@@ -2481,23 +2481,12 @@ where
         self.emit_leading_comments_of_span(node.span(), false)?;
 
         srcmap!(node, true);
-
-        let is_last_rest = match node.props.last() {
-            Some(ObjectPatProp::Rest(..)) => true,
-            _ => false,
-        };
-        let format = if is_last_rest {
-            ListFormat::ObjectBindingPatternElements ^ ListFormat::AllowTrailingComma
-        } else {
-            ListFormat::ObjectBindingPatternElements
-        };
```

Comment:
Ideally this logic should be in `emit_list` or in parent, not in `ObjectPat` itself, because `ObjectPat` doesn't know parent context